src/middleware/packages/triplestore/subservices/dataset.js

```diff
@@ -103,6 +114,48 @@ const DatasetService = {
           task = await response.json();
         }
       } while (!task || !task.finished);
+    },
+    delete: {
+      params: {
+        dataset: { type: 'string' },
+        iKnowWhatImDoing: { type: 'boolean' }
+      },
+      async handler(ctx) {
+        const { dataset, iKnowWhatImDoing } = ctx.params;
+        if (!iKnowWhatImDoing) {
+          throw new Error('Please confirm that you know what you are doing by setting `iKnowWhatImDoing` to `true`.');
+        }
+        const isSecure = await this.actions.isSecure({ dataset });
+
+        if (isSecure && !this.settings.fusekiBase)
+          throw new Error(
+            'Please provide the fusekiBase dir setting to the triplestore service, to delete a secure dataset.'
+          );
+
+        const response = await fetch(urlJoin(this.settings.url, '$/datasets', dataset), {
+          method: 'DELETE',
+          headers: this.headers
+        });
+        if (!response.ok) {
+          throw new Error(`Failed to delete dataset ${dataset}: ${response.statusText}`);
+        }
```

Comment:
Shouldn't we run this code only if the dataset is not secure ?

Reply:
I think both options are viable.
By calling it anyways, we ensure that the dataset is not queriable anymore  (will throw an error).
If we don't call it, the dataset remains in memory until the fuseki server restarts.